libebpfdiscoveryskel/src/TrackedSession.h

```diff
@@ -50,12 +53,142 @@ struct {
+	}
+	return sockIpPtr;
+}
+
+__attribute__((always_inline)) inline static int setSockIPv4ForTrackedSession(
+		const struct DiscoveryTrackedSessionKey* key, __u64 pidTgid, struct DiscoverySockIPv4* sockIpPtr) {
+	int ret = bpf_map_update_elem(&trackedSessionSockIPv4Map, key, sockIpPtr, BPF_ANY);
+	if (ret != 0) {
+		return ret;
+	}
+	return bpf_map_update_elem(&trackedSessionSockIPv4PidTgidMap, &pidTgid, sockIpPtr, BPF_ANY);
+}
+
+__attribute__((always_inline)) inline static int setSockIPv6ForTrackedSession(
+		const struct DiscoveryTrackedSessionKey* key, __u64 pidTgid, struct DiscoverySockIPv6* sockIpPtr) {
+	int ret = bpf_map_update_elem(&trackedSessionSockIPv6Map, key, sockIpPtr, BPF_ANY);
+	if (ret != 0) {
+		return ret;
+	}
+	return bpf_map_update_elem(&trackedSessionSockIPv6PidTgidMap, &pidTgid, sockIpPtr, BPF_ANY);
+}
+
+__attribute__((always_inline)) inline static int deleteSockIPv4ForTrackedSession(
+		const struct DiscoveryTrackedSessionKey* key, __u64 pidTgid) {
+	return bpf_map_delete_elem(&trackedSessionSockIPv4Map, key);
+}
+
+__attribute__((always_inline)) inline static int deleteSockIPv6ForTrackedSession(
+		const struct DiscoveryTrackedSessionKey* key, __u64 pidTgid) {
+	return bpf_map_delete_elem(&trackedSessionSockIPv6Map, key);
+}
+
+__attribute__((always_inline)) inline static int deleteTrackedSession(struct DiscoveryTrackedSessionKey* key, __u64 pidTgid) {
+	int ret = bpf_map_delete_elem(&trackedSessionsMap, key);
+	deleteSockIPv4ForTrackedSession(key, pidTgid);
+	deleteSockIPv6ForTrackedSession(key, pidTgid);
+	return ret;
+}
+
+__attribute__((always_inline)) inline static int createTrackedSessionUnencryptedHttp(
+		struct DiscoveryAllSessionState* allSessionStatePtr, struct DiscoveryTrackedSessionKey* key) {
+	struct DiscoverySession session = {
+			.bufferCount = 0, .id = ++allSessionStatePtr->sessionCounter, .flags = DISCOVERY_FLAG_SESSION_UNENCRYPTED_HTTP};
+	return bpf_map_update_elem(&trackedSessionsMap, (struct DiscoveryTrackedSessionKey*)key, &session, BPF_ANY);
+}
+
+__attribute__((always_inline)) inline static int createTrackedSessionSslHttp(
+		struct DiscoveryAllSessionState* allSessionStatePtr, struct DiscoveryTrackedSessionKey* key) {
+	struct DiscoverySession session = {
+			.bufferCount = 0, .id = ++allSessionStatePtr->sessionCounter, .flags = DISCOVERY_FLAG_SESSION_SSL_HTTP};
+	return bpf_map_update_elem(&trackedSessionsMap, (struct DiscoveryTrackedSessionKey*)key, &session, BPF_ANY);
+}
+
+__attribute__((always_inline)) inline static int trackedSessionPutSourceIp(
+		struct DiscoverySession* sessionPtr, struct DiscoverySavedBufferKey* key, __u64 pidTgid) {
+	void* sockIpPtr = getSockIPv4ForTrackedSession((struct DiscoveryTrackedSessionKey*)key, pidTgid);
+	if (sockIpPtr != NULL) {
+		discoveryFlagsSessionSetIPv4(&sessionPtr->flags);
+		BPF_CORE_READ_INTO(&sessionPtr->sourceIP.data, (struct sockaddr_in*)sockIpPtr, sin_addr);
+		bpf_map_delete_elem(&trackedSessionSockIPv4Map, key);
+		return 0;
```

Comment:
please use `EXIT_SUCCESS` / `EXIT_FAILURE`

Reply:
Those are constants from libc that we don't use in BPF code.